dragonfly/room2d.py

```diff
@@ -1061,6 +1134,90 @@ def _flip_wall_assigned_objects(original_geo, bcs, win_pars, shd_pars):
         # retrun the flipped lists
         return new_bcs, new_win_pars, new_shd_pars
 
+    def _split_plenum_walls_along_height(self, hb_room, tolerance, floor_height,
+                                         floor_to_ceiling_height):
+        """
+        Args:
+            plenum_hb_room: A plenum represented as a Honeybee Room.
+        """
+        # Set ceiling height
+        ceiling_height = floor_height + floor_to_ceiling_height
+
+        # Iterate through bc of parent room
+        new_faces = [hb_room[0]]
+        for i, bc in enumerate(self._boundary_conditions):
+            face = hb_room[i + 1]
+            if not isinstance(bc, Surface):
+                new_faces.append(face)
+            else:
+                adj_rm = self._parent.room_by_identifier(
+                    bc.boundary_condition_objects[-1])
+                flr_diff = adj_rm.floor_height - floor_height
+                ciel_diff = ceiling_height - adj_rm.ceiling_height
+
+                # Use diffs to split the wall
+                if flr_diff <= tolerance and ciel_diff <= tolerance:
+                    # No need to split the surface along its height
+                    new_faces.append(face)
+                elif flr_diff > tolerance and ciel_diff > tolerance:
+                    # split the face into to 3 smaller faces along its height
```

Comment:
As I understand it, this case really shouldn't be happening for a plenum (a plenum would only ever be split in two and not three). Or I guess you have this case because you exposed `plenum_ht` as an input. Let's get rid of that input and then we can always assume that all plenums are going to the same height.

Reply:
As discussed above, I'll get rid of this for now and build up the more important plenum features first (i.e getting BCs and geometry working).